index.html

```diff
@@ -1787,6 +1935,296 @@ <h2>
+        with [=file type=] on the operating system.
+      </p>
+      <p>
+        A <dfn>file type</dfn> can be defined by a [=mime type=] and/or [=file
+        extension=]. A file belongs to a file type if the OS determines it to
+        have a mime type and/or its name ends with a certain [=file
+        extension=]. A <dfn>file extension</dfn> is a string that start with a
+        "." and only contains [=valid suffix code points=]. A <dfn>valid suffix
+        code point</dfn> is a [=code point=] that is [=ASCII alphanumeric=],
+        U+002B (+), or U+002E (.). Additionally, [=file extensions=] are are
+        limited to a length of 16 code points.
+      </p>
+      <aside class="note">
+        <p>
+          Note: These code points were chosen to support most pre-existing file
+          formats. The vast majority of file extensions are purely
+          alphanumeric, but compound extensions (such as .tar.gz) and
+          extensions such as .c++ for C++ source code are also fairly common,
+          hence the inclusion of + and . as allowed code points.
+        </p>
+      </aside>
+      <aside class="note">
+        <p>
+          The selection of an app to handle a file launch if there are multiple
+          apps registerd for the file's [=file type=] is functionality left to
+          the operating system.
+        </p>
+      </aside>
+      <section>
+        <h3>
+          `action` member
+        </h3>
+        <p>
+          The [=file handler=]'s <code><dfn data-dfn-for=
+          "file handler">action</dfn></code> member is a <a>string</a> that
+          represents a relative URL of the [=manifest/start_url=] origin that
+          is [=manifest/within scope=] of a [=Document/processed manifest=].
+          This URL will be navigated to in the steps to [=execute a file
+          handler launch=].
+        </p>
+      </section>
+      <section>
+        <h3>
+          `name` member
+        </h3>
+        <p>
+          The [=file handler=]'s <code><dfn data-dfn-for=
+          "file handler">name</dfn></code> member is a <a>string</a> that
+          represents a name to be provided to the user's operating system for
+          use with "Open with..." UX flows. This MAY be replaced by the user
+          agent for privacy & security reasons.
+        </p>
+      </section>
+      <section data-cite="MIMETYPE">
+        <h3>
+          `accept` member
+        </h3>
+        <p>
+          The [=file handler=]'s <code><dfn data-dfn-for=
+          "file handler">accept</dfn></code> member is a <a>dictionary</a>
+          mapping [=MIME types=] to a list of extensions. Extensions have to be
+          strings that start with a "." and only contain [=valid suffix code
+          points=]. Additionally extensions are limited to a length of 16 code
+          points.
+        </p>
+        <p>
+          In addition to complete [=MIME types=], "*" can be used as the
```

Comment:
While the wildcard in the subtype would work in some OSes, on others only the extension matters. Thus it's probably worth noting that

`"image/*": [ ".png" ]`

would /not/ handle all image files on all OSes. (In fact there's no enforcement of these two lists having anything to do with each other, so

`"image/png": [ ".txt" ]`

would be accepted; it would just behave differently across OSes.